docs/writing-docs/mdx.md

```diff
@@ -60,190 +42,181 @@ For example, here's the first story from the Checkbox example above, rewritten i
 
-<CodeSnippets
-  paths={[
-    'common/component-story-mdx-decorators-params.mdx.mdx',
-  ]}
-/>
+```jsx
+<Guidelines>
+  <Dos>
+    - Use buttons for the main actions on your page - Identify the primary action and make that
+    `primary`
+  </Dos>
+  <Donts>
+    - Use a button when a link will do (e.g. for navigation only actions) - Use multiple primary
+    buttons in a single UI state
+  </Donts>
+</Guidelines>
+```
 
-<!-- prettier-ignore-end -->
+### Known limitations
 
-Global parameters and decorators work just like before.
+- While MDX2 supports a variety of runtimes (React, Preact, Vue), Storybook’s implementation is React-only. That means that your documentation is rendered in React, while your stories are rendered in the runtime of your choice (React, Vue, Angular, Web-components, Svelte, etc.)
 
-## Documentation-only MDX
+## Breaking changes
 
-Typically, when you use Storybook MDX, you define stories in the MDX, and documentation is automatically associated with those stories. But what if you want to write Markdown-style documentation without any stories inside?
+There are a lot of breaking changes if you’re moving from MDX1 to MDX2. As far as we know, all of these are due to changes in the MDX library itself, rather than changes to Storybook’s usage. Nevertheless, as an MDX user, you will probably need to update your MDX files as part of the upgrade. MDX has published their own [Migration guide](https://mdxjs.com/migrating/v2/#update-mdx-files). Here we try to summarize some of the key changes for Storybook users.
 
-Suppose you don't define stories in your MDX. In that case, you can write MDX documentation and associate it with an existing story or embed that MDX as its own documentation node in your Storybook's navigation.
+### Custom components apply differently
 
-If you don't define a Meta, you can write Markdown and associate with an existing story. See ["CSF Stories with MDX Docs"](../../addons/docs/docs/recipes.md#csf-stories-with-mdx-docs).
+From the MDX migration guide:
 
-To get a "documentation-only story" in your UI, define a `<Meta>` as you normally would, but don't define any stories. It will show up in your UI as a documentation node:
+> We now “sandbox” components, for lack of a better name. It means that when you pass a component for `h1`, it does get used for `# hi` but not for `<h1>hi</h1>`
 
-![MDX docs only story](./mdx-documentation-only.png)
+This means that the first heading in the following example gets replaced, whereas the second heading does not. This doesn’t sound like a big change, but in practice is extremely disruptive and manifests itself in a variety of ways. Unfortunately, this cannot be automatically converted in a safe way.
 
-### Syntax highlighting
```

Comment:
This was merged with https://storybook.js.org/docs/7.0/react/faq#why-arent-my-code-blocks-highlighted-with-storybook-mdx